File: TheLionsDen.Model/Requests/ReservationInsertRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace TheLionsDen.Model.Requests
{
    public class ReservationInsertRequest
    {
        [Required]
        public DateTime Arrival { get; set; }
        [Required]
        public DateTime Departure { get; set; }
        [Required]
        public float TotalPrice { get; set; }
        [MaxLength(40)]
        public string EstimatedArrivalTime { get; set; }
        [Required]
        public int UserId { get; set; }
        [Required]
        public int RoomId { get; set; }
        [Required]
        public PaymentDetailsInsertRequest PaymentDetails { get; set; }

        public List<int> FacilityIds { get; set; }
    }
}

File: the_lions_den_mobile/lib/model/reservation/reservation_insert_request.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:the_lions_den_mobile/model/reservation/payment_details_insert_request.dart';
part 'reservation_insert_request.g.dart';

@JsonSerializable()
class ReservationInsertRequest {
  DateTime? arrival;
  DateTime? departure;
  double? totalPrice;
  String? estimatedArrivalTime;
  int? userId;
  int? roomId;
  PaymetDetailsInsertRequest? paymentDetails;
  List<int>? facilityIds;

  ReservationInsertRequest();

  factory ReservationInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$ReservationInsertRequestFromJson(json);

  /// Connect the generated [_$ReservationInsertRequestToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$ReservationInsertRequestToJson(this);
}
